Snowflake.Data.Tests/IntegrationTests/SFConnectionWithTomlIT.cs

```diff
@@ -0,0 +1,135 @@
+                {
+                    conn.Open();
+                    Assert.AreEqual(ConnectionState.Open, conn.State);
+                }
+            }
+            finally
+            {
+                Environment.SetEnvironmentVariable(TomlConnectionBuilder.SnowflakeHome, snowflakeHome);
+            }
+        }
+
+        [Test]
+        public void TestThrowExceptionIfTomlNotFoundWithOtherConnectionString()
+        {
+            var snowflakeHome = Environment.GetEnvironmentVariable(TomlConnectionBuilder.SnowflakeHome);
+            var connectionName = Environment.GetEnvironmentVariable(TomlConnectionBuilder.SnowflakeDefaultConnectionName);
+            CreateTomlConfigBaseOnConnectionString(ConnectionString);
+            Environment.SetEnvironmentVariable(TomlConnectionBuilder.SnowflakeHome, s_workingDirectory);
+            Environment.SetEnvironmentVariable(TomlConnectionBuilder.SnowflakeDefaultConnectionName, "notfoundconnection");
+            try
+            {
+                using (var conn = new SnowflakeDbConnection())
+                {
+                    Assert.Throws<SnowflakeDbException>(() => conn.Open(), "Unable to connect. Specified connection name does not exist in connections.toml");
+                }
+            }
+            finally
+            {
+                Environment.SetEnvironmentVariable(TomlConnectionBuilder.SnowflakeHome, snowflakeHome);
+                Environment.SetEnvironmentVariable(TomlConnectionBuilder.SnowflakeDefaultConnectionName, connectionName);
+            }
+        }
+
+        [Test]
+        public void TestThrowExceptionIfTomlFromNotFoundFromDbConnection()
+        {
+            var snowflakeHome = Environment.GetEnvironmentVariable(TomlConnectionBuilder.SnowflakeHome);
+            Environment.SetEnvironmentVariable(TomlConnectionBuilder.SnowflakeHome, s_workingDirectory);
+            try
+            {
+                using (var conn = new SnowflakeDbConnection())
+                {
+                    Assert.Throws<SnowflakeDbException>(() => conn.Open(), "Error: Required property ACCOUNT is not provided");
+                }
+            }
+            finally
+            {
+                Environment.SetEnvironmentVariable(TomlConnectionBuilder.SnowflakeHome, snowflakeHome);
+            }
+        }
+
+        private static void CreateTomlConfigBaseOnConnectionString(string connectionString)
+        {
+            var tomlModel = new TomlTable();
+            var properties = SFSessionProperties.ParseConnectionString(connectionString, null);
+
+            var defaultTomlTable = new TomlTable();
+            tomlModel.Add("default", defaultTomlTable);
+
+            foreach (var property in properties)
+            {
+                defaultTomlTable.Add(property.Key.ToString(), property.Value);
+            }
+
+            var filePath = Path.Combine(s_workingDirectory, "connections.toml");
+            using (var writer = File.CreateText(filePath))
```

Comment:
maybe we could create an empty file first, then change the permissions for this file and then append the content, and then maybe once more time change the permissions (just to be sure)?